cspann: switch tests to use new fixups

andy-kimball opened the pull request:
Now that incremental split is working, switch all C-SPANN tests to use
it. In the near term, this will reduce search accuracy, since the new
algorithm doesn't perform vector movement across partitions. Also,
incremental merge is not yet implemented, so certain tests will not
behave as originally expected.

Epic: CRDB-42943

Release note: None

andy-kimball committed:
memstore: maintain stats in TryUpdatePartitionMetadata

TryUpdatePartitionMetadata is used to add/remove levels from the K-means tree.
Change it to grow/shrink CVStats when a level is added or removed.

Epic: CRDB-42943

Release note: None

andy-kimball committed:
memstore: serialize partition metadata in memstore

Serialize partition metadata when saving a MemStore instance to disk.
This is used for testing.

Epic: CRDB-42943

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144036)
<!-- Reviewable:end -->

andy-kimball committed:
cspann: switch tests to use new fixups

Now that incremental split is working, switch all C-SPANN tests to use
it. In the near term, this will reduce search accuracy, since the new
algorithm doesn't perform vector movement across partitions. Also,
incremental merge is not yet implemented, so certain tests will not
behave as originally expected.

Epic: CRDB-42943

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 2 of 2 files at r2, 18 of 18 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144036#-:-ONLOt_zBSflUTRUukj_:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mw5h)

___
*[`pkg/sql/vecindex/cspann/memstore/memstore.go` line 471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/144036#-ONLLJo64ghyjV95PC4p:-ONLLJo64ghyjV95PC4q:b-abgofv) ([raw file](https://github.com/cockroachdb/cockroach/blob/fc0402da64eabe5a74514b85a68819df462df4f8/pkg/sql/vecindex/cspann/memstore/memstore.go#L471)):*
> ```Go
> 		memPart.lock.created = s.tickLocked()
> 
> 		// Grow or shrink CVStats slice.
> ```

nit: make note that the stats are for non-leaf levels, so we subtract one to get the new slice length.
___
*[`pkg/sql/vecindex/cspann/index_test.go` line 77 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/144036#-ONLORCR01KBcXt8AQLQ:-ONLORCR01KBcXt8AQLR:b-o1aqgz) ([raw file](https://github.com/cockroachdb/cockroach/blob/e2c20c366d7c77f9b50ab4f7a5b8c305803d9181/pkg/sql/vecindex/cspann/index_test.go#L77)):*
> ```Go
> 
> 				case "skip-fixups":
> 					// Skip processing of any fixups generated by the command.
> ```

This is different from `discard-fixups` in that a subsequent command can trigger fixup processing?


<!-- Sent from Reviewable.io -->

andy-kimball commented:
bors r=drewkimball
